pkg/controller/traffic/pod_label_shifter.go

```diff
@@ -18,159 +18,168 @@ import (
 
 type podLabelShifter struct {
 	appName               string
+	releaseName           string
 	namespace             string
-	serviceSelector       labels.Selector
 	clusterReleaseWeights clusterReleaseWeights
 }
 
 type clusterReleaseWeights map[string]map[string]uint32
 
 func newPodLabelShifter(
 	appName string,
+	releaseName string,
 	namespace string,
 	trafficTargets []*shipper.TrafficTarget,
 ) (*podLabelShifter, error) {
-
 	weights, err := buildClusterReleaseWeights(trafficTargets)
 	if err != nil {
 		return nil, err
 	}
 
-	serviceSelector := map[string]string{
-		shipper.AppLabel: appName,
-		shipper.LBLabel:  shipper.LBForProduction,
-	}
-
 	return &podLabelShifter{
 		appName:               appName,
+		releaseName:           releaseName,
 		namespace:             namespace,
-		serviceSelector:       labels.Set(serviceSelector).AsSelector(),
 		clusterReleaseWeights: weights,
 	}, nil
 }
 
 func (p *podLabelShifter) SyncCluster(
 	cluster string,
-	release string,
 	clientset kubernetes.Interface,
 	informerFactory kubeinformers.SharedInformerFactory,
 ) (uint32, error) {
-	releaseWeights, ok := p.clusterReleaseWeights[cluster]
+	releaseTargetWeights, ok := p.clusterReleaseWeights[cluster]
 	if !ok {
 		return 0, shippererrors.NewMissingTrafficWeightsForClusterError(
 			p.namespace, p.appName, cluster)
 	}
 
-	appSelector := labels.Set{shipper.AppLabel: p.appName}.AsSelector()
-	releaseSelector := labels.Set{shipper.ReleaseLabel: release}.AsSelector()
-	trafficSelector, err := p.getTrafficSelector(cluster, informerFactory)
-	if err != nil {
-		return 0, err
-	}
-
 	podLister := informerFactory.Core().V1().Pods().Lister().Pods(p.namespace)
 	podGVK := corev1.SchemeGroupVersion.WithKind("Pod")
 
+	appSelector := labels.Set{shipper.AppLabel: p.appName}.AsSelector()
 	appPods, err := podLister.List(appSelector)
 	if err != nil {
 		return 0, shippererrors.NewKubeclientListError(
 			podGVK, p.namespace, appSelector, err)
 	}
 
-	releasePods, err := podLister.List(releaseSelector)
+	releaseSelector := labels.Set{shipper.ReleaseLabel: p.releaseName}.AsSelector()
+
+	svc, err := p.getService(informerFactory)
```

Comment:
This implies on a singular active service per release, which is very true now. Might be worth leaving a comment this assumption was kept in mind. If this constraint would be challenged in the future, we should remember about this spot.